openfl/federated/data/loader_xgb.py

```diff
@@ -0,0 +1,168 @@
+                (default is None).
+            num_batches (int, optional): The number of batches for the data
+                loader (default is None).
+
+        Returns:
+            generator: The generator object for the training data.
+        """
+        return self._get_batch_generator(
+            X=self.X_train,
+            y=self.y_train,
+            batch_size=batch_size,
+            num_batches=num_batches,
+        )
+
+    def get_valid_loader(self, batch_size=None):
+        """Returns the data loader for the validation data.
+
+        Args:
+            batch_size (int, optional): The batch size for the data loader
+                (default is None).
+
+        Returns:
+            generator: The generator object for the validation data.
+        """
+        return self._get_batch_generator(X=self.X_valid, y=self.y_valid, batch_size=batch_size)
+
+    def get_train_data_size(self):
+        """Returns the total number of training samples.
+
+        Returns:
+            int: The total number of training samples.
+        """
+        return self.X_train.shape[0]
+
+    def get_valid_data_size(self):
+        """Returns the total number of validation samples.
+
+        Returns:
+            int: The total number of validation samples.
+        """
+        return self.X_valid.shape[0]
+
+    @staticmethod
+    def _batch_generator(X, y, idxs, batch_size, num_batches):
+        """Generates batches of data.
+
+        Args:
+            X (np.array): The input data.
+            y (np.array): The label data.
+            idxs (np.array): The index of the dataset.
+            batch_size (int): The batch size for the data loader.
+            num_batches (int): The number of batches.
+
+        Yields:
+            tuple: The input data and label data for each batch.
+        """
+        for i in range(num_batches):
+            a = i * batch_size
+            b = a + batch_size
+            yield X[idxs[a:b]], y[idxs[a:b]]
+
+    def _get_batch_generator(self, X, y, batch_size, num_batches=None):
+        """Returns the dataset generator.
+
+        Args:
+            X (np.array): The input data.
+            y (np.array): The label data.
+            batch_size (int): The batch size for the data loader.
+            num_batches (int, optional): The number of batches (default is
+                None).
+
+        Returns:
+            generator: The dataset generator.
+        """
+        if batch_size is None:
+            batch_size = self.batch_size
+
+        # shuffle data indices
+        if self.random_seed is not None:
+            np.random.seed(self.random_seed)
+
+        idxs = np.random.permutation(np.arange(X.shape[0]))
+
+        # compute the number of batches
+        if num_batches is None:
+            num_batches = ceil(X.shape[0] / batch_size)
+
+        # build the generator and return it
+        return self._batch_generator(X, y, idxs, batch_size, num_batches)
```

Comment:
It would improve readability if you moved the private utility methods at the bottom.

PS: not a blocker for this PR, but as also pointed out by @MasterSkepticista, it's unfortunate that those utilities have to be copy/pasted across all data loader base classes for the supported ML frameworks. If there's already a utility module that all those classes can import, such methods should go there. Otherwise we should create one. WDYT?

Reply:
As an alternative, I've removed the batching and dataloader methods since they're not relevant to XGBoost

I do think it would be good to include a utility module for methods that are reused on multiple places. I won't include it in this PR, but it should be added to the backlog since it could clean up a lot of the code in other loaders/runners

Reply:
Sounds good, thanks!